packages/runtime/runtime-utils/src/serializer.ts

```diff
@@ -27,103 +130,38 @@ export class FluidSerializer implements IFluidSerializer {
-                    return value;
-                }
-
-                // Old documents may have handles with relative path in their summaries. Convert these to absolute
-                // paths. This will ensure that future summaries will have absolute paths for these handles.
-                const absolutePath = value.url.startsWith("/")
-                    ? value.url
-                    : generateHandleContextPath(value.url, this.context);
-                return new RemoteFluidObjectHandle(absolutePath, this.root);
-            });
-    }
-
-    // Invoked by `replaceHandles()` for non-null objects to recursively replace IFluidHandle references
-    // with serialized handles (cloning as-needed to avoid mutating the original `input` object.)
-    private recursivelyReplaceHandles(
-        input: any,
-        bind: IFluidHandle,
-    ) {
-        // If the current input is an IFluidHandle instance, replace this leaf in the object graph with
-        // the handle's serialized from.
-
-        // Note: Caller is responsible for ensuring that `input` is a non-null object.
-        const handle = input.IFluidHandle;
-        if (handle !== undefined) {
-            return this.serializeHandle(handle, bind);
-        }
-
-        // eslint-disable-next-line @typescript-eslint/ban-types
-        let clone: object | undefined;
-        for (const key of Object.keys(input)) {
-            const value = input[key];
-            // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
-            if (!!value && typeof value === "object") {
-                // Note: Except for IFluidHandle, `input` must not contain circular references (as object must
-                //       be JSON serializable.)  Therefore, guarding against infinite recursion here would only
-                //       lead to a later error when attempting to stringify().
-                const replaced = this.recursivelyReplaceHandles(value, bind);
-
-                // If the `replaced` object is different than the original `value` then the subgraph contained one
-                // or more handles.  If this happens, we need to return a clone of the `input` object where the
-                // current property is replaced by the `replaced` value.
-                if (replaced !== value) {
-                    // Lazily create a shallow clone of the `input` object if we haven't done so already.
-                    clone = clone ?? (Array.isArray(input)
-                        ? [...input]
-                        : { ...input });
-
-                    // Overwrite the current property `key` in the clone with the `replaced` value.
-                    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-                    clone![key] = replaced;
-                }
-            }
-        }
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
```

Comment:
'recursivelyReplaceHandles()' lifted to new 'recursivelyReplace' helper at beginning.